ApiReview/API.Polly.Core/NoDocs/Polly/ResiliencePipeline.cs

```diff
@@ -0,0 +1,36 @@
+// Assembly 'Polly.Core'
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Polly.Utils;
+
+namespace Polly;
+
+public sealed class ResiliencePipeline
```

Comment:
Make `IDisposable`,  `IAsyncDisposable`

- if pipeline is created explicitly from builder disposal is allowed
- if pipeline has a parent (registry) the dispose is not allowed, only when the registry is disposed it disposes all it's pipelines